lib/vmdb/loggers.rb

```diff
@@ -76,6 +80,37 @@ def self.create_logger(log_file_name, logger_class = VMDBLogger)
       nil
     end
 
+    private_class_method def self.progname_from_file(log_file_name)
+      File.basename(log_file_name, ".*")
+    end
+
+    private_class_method def self.create_wrapper_logger(log_file_name, logger_class, wrapped_logger)
+      log_file_name = Pathname.new(log_file_name) unless log_file_name.kind_of?(Pathname)
+      progname      = progname_from_file(log_file_name)
+
+      logger_class.new(nil, progname: progname).tap do |logger|
+        # HACK: In order to access the "filename" of the wrapped logger, we inject it as an instance var.
+        logger.instance_variable_set(:@log_file_name, log_file_name)
```

Comment:
Is this used anywhere else?  I don't see any callers outside of the current `self.create_logger(log_file_name, logger_class = VMDBLogger)`

Reply:
3 lines down it's exposed as the content of the filename method.